tests/parametric/test_span_links.py

```diff
@@ -0,0 +1,189 @@
+                with test_library.start_span(
+                    "child",
+                    parent_id=parent.span_id,
+                    links=[Link(parent_id=parent.span_id, attributes={"foo": "bar", "array": ["a", "b", "c"]})],
+                ):
+                    pass
+
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, Span(name="root"))
+        assert len(trace) == 2
+
+        root = root_span(trace)
+        span = find_span(trace, Span(name="child"))
+
+        assert span.get("parent_id") == root.get("span_id")
+        assert span.get("span_links") is not None
+
+        span_links = span["span_links"]
+        assert len(span_links) == 1
+        link = span_links[0]
+        assert link.get("span_id") == root.get("span_id")
+        assert link.get("trace_id") == root.get("trace_id")
+        assert (link.get("trace_id_high") or 0) == 0
+        assert link["attributes"].get("foo") == "bar"
+        assert link["attributes"].get("array.0") == "a"
+        assert link["attributes"].get("array.1") == "b"
+        assert link["attributes"].get("array.2") == "c"
+        assert (link.get("flags") or 0) == 0
+
+    def test_span_link_from_distributed_datadog_headers(self, test_agent, test_library):
+        """Properly inject datadog distributed tracing information into span links.
+        Testing the conversion of x-datadog-* headers to tracestate for
+        representation in span links.
+        """
+        with test_library:
+            with test_library.start_span(
+                "root",
+                links=[Link(parent_id=0, attributes={"foo": "bar"})],
+                http_headers=[
+                    ["x-datadog-trace-id", "1234567890"],
+                    ["x-datadog-parent-id", "9876543210"],
+                    ["x-datadog-sampling-priority", "2"],
+                    ["x-datadog-origin", "synthetics"],
+                    ["x-datadog-tags", "_dd.p.dm=-4,_dd.p.tid=0000000000000010"],
+                ],
+            ):
+                pass
+
+        span = get_span(test_agent)
+        assert span.get("trace_id") != 1234567890
+        assert span_has_no_parent(span)
+        assert span["meta"].get(ORIGIN) is None
+        assert span["meta"].get("_dd.p.dm") != "-4"
+        assert span["metrics"].get(SAMPLING_PRIORITY_KEY) != 2
+
+        assert span.get("span_links") is not None
+
+        span_links = span["span_links"]
+        assert len(span_links) == 1
+        link = span_links[0]
+        assert link.get("span_id") == 9876543210
+        assert link.get("trace_id") == 1234567890
+        assert link.get("trace_id_high") == 16
+
+        assert link.get("tracestate") is not None
+        tracestateArr = link["tracestate"].split(",")
+        assert len(tracestateArr) == 1 and tracestateArr[0].startswith("dd=")
+        tracestateDD = tracestateArr[0][3:].split(";")
+        assert len(tracestateDD) == 3
```

Comment:
Should it be at least 3? If it will prevent the test to break when adding other fields later.

Reply:
Don't know, it also allows to catch if ... you accidentally put distributed tags into span link attributes for example. (I did that in PHP, oops.)

Reply:
Both options are good. Let's keep it that way then :)